src/c++/perf_analyzer/report_writer.cc

```diff
@@ -209,7 +216,15 @@ ReportWriter::GenerateReport()
           ofs << avg_latency_us << ",";
         }
         ofs << std::to_string(avg_send_time_us + avg_receive_time_us) << ",";
-        ofs << std::to_string(avg_response_wait_time_us);
+        ofs << std::to_string(avg_response_wait_time_us) << ",";
+        if (should_collect_metrics_) {
+          if (status.metrics.size() == 1) {
+            WriteGpuMetrics(ofs, status.metrics[0]);
+          } else {
+            throw std::runtime_error(
+                "There should only be one entry in the metrics vector.");
```

Comment:
Does anybody catch this exception?

Reply:
Nope. It propagates up to `main()` and ends the program. It's more of a signal to the developer that they've called this function incorrectly.

Reply:
I think it would look awkward if this happens. It will contain the full stackstrace which I don't think is the best UX.

Reply:
What other behavior do you propose? It's an invalid input to the function. Should the function not validate its inputs and risk accessing an element that doesn't exist or risk not informing developer of misusing the vector by putting more than one element in it?

Reply:
I think this is a fallout of the fact that the data structure we use at this point of the code isn't ideally designed. It really shouldn't be a vector here but we're reusing a struct data member for both the profiling phase and the finalized-aggregated-report phase. A future ticket for refactoring would be a good idea.

Reply:
I'm just saying to catch this exception in main and print an error and exit with the appropriate exit code. Or just return an error object so that it will be handled properly by the existing code.

Reply:
Resolving based on the discussion offline.